### ISSUE TITLE
blueprints - import@ always on top

### ISSUE THREAD

**Issue by Genenenenaam** [getgrav/grav#1844]
When [Embedding Form (@import)](https://learn.getgrav.org/forms/blueprints/advanced-features#embedding-form-import), the imported fields always go on top, above the default fields. Using ordering@ has no effect.

(SEE SCREENSHOT)
<img width="1452" alt="schermafbeelding 2018-01-28 om 22 06 05" src="https://user-images.githubusercontent.com/20650976/35487049-8db909e0-0477-11e8-9b42-842dfd3f45a5.png">

The parent file:
`title: Product
extends@: default

form:
  fields:
    tabs:
      type: tabs
      active: 1

      fields:
        content:
          type: tab
          title: Content

          #
          # Add your custom fields here.
          # These fields will be available in the page's Content tab
          # inside the Admin Panel.
          #
          fields:

            #
            # Hero
            #
            import@: fields/hero`

the fields to import:
`form:
  #
  # Hero
  #
  Hero:
    type: section
    title: Hero
    underline: true

  # Hero title
  header.heroTitle:
    type: text
    label: A title
    validate: 
      required: true`

How do you keep the default fields on top?
The [documentation](https://learn.getgrav.org/forms/blueprints/advanced-features#changing-field-ordering) states: 

> When you extend a blueprint or import a file, by default the new fields are added to the end of the list.

But this seems not to be the case? 

What am i doing wrong here?

**Comment by rhukster** [comment]
This is currently a limitation of the import@ feature.  I've run into this myself, and currently, there is not work around to fix it.

**Comment by rhukster** [comment]
@mahagr do you have any thoughts on how to address this?

**Comment by mahagr** [comment]
Custom ordering should work if the blueprint is properly initialized. Will need to investigate this.

**Comment by mahagr** [comment]
You're right, the current logic doesn't allow you to append the blueprints where you want them to be.

I've been working on adding support for multiple imports in the same level:

```yaml
import@1: fields/hero
import@2: fields/footer
```

I've also added support for partial imports:

```yaml
import@: fields/hero:hero
import@: fields/footer:test/fields
```

I'm still trying to figure out how to fix the ordering...

**Comment by mahagr** [comment]
See https://github.com/rockettheme/toolbox/issues/17

**Comment by mahagr** [comment]
Your issue is in https://github.com/rockettheme/toolbox/issues/18

**Comment by sebastianbaumann** [comment]
Any news on multiple import feature? I've just stumbled accross this, when I tried to import multiple partials into one blueprint file.

**Comment by mahagr** [comment]
@sebastianbaumann Use 

```yaml
          fields:
            import@1: partials/cta
            import@2: partials/fileupload
```

**Comment by nerikh** [comment]
@mahagr can you confirm using import@1 ... import@2 is a very recent addition as I tried that a few weeks ago and it was not working.

**Comment by mahagr** [comment]
Looks like this feature isn't yet in the released `rockettheme/toolbox` package. You can try it by changing `composer.json` file and running `composer update --no-dev` (you may need to install composer first, google it):

```
        "rockettheme/toolbox": "dev-develop",
```

Find the line and replace the value before running the command.

**Comment by nerikh** [comment]
Thank you, I can work with composer but may wait until it's in grav core.

**Comment by NicoHood** [comment]
This works for me in grav 1.6.28, I guess the issue can be closed now.

**Comment by mahagr** [comment]
Number is there just do make the key to differ, allowing multiple instances of the same command.

**Comment by NicoHood** [comment]
So sorting is not possible at all? :-(

I think the issue can be closed then, as it works as expected. Is implementing a sorting function an option to you? We could track it in another issue.

**Comment by mahagr** [comment]
Yeah... Unfortunately the blueprints do not handle importing other documents too well. However, if you define the structure in the main template, you will be able to dictate the ordering of the main elements.
